packages/context/test/acceptance/extension-point.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Context, Binding, BindingScope, inject} from '../..';
+
+// tslint:disable:no-any
+/**
+ * Interface for the extension point configuration
+ */
+export interface ExtensionPointConfig {
+  extensions: {
+    [extensionName: string]: any;
+  };
+  extensionPoint: {
+    [property: string]: any;
+  };
+}
+
+/**
+ * Base class for extension points
+ */
+export abstract class ExtensionPoint<EXT extends object> {
+  /**
+   * Configuration (typically to be injected)
+   */
+  protected configuration: ExtensionPointConfig;
+
+  constructor(
+    /**
+     * The unique name of this extension point. It also serves as the binding
+     * key prefix for bound extensions
+     */
+    protected name: string,
+    /**
+     * The Context (typically to be injected)
+     */
+    protected context: Context,
+    /**
+     * Configuration (typically to be injected)
+     */
+    configuration?: ExtensionPointConfig,
+  ) {
+    this.configuration = configuration || {extensionPoint: {}, extensions: {}};
+  }
+
+  /**
+   * Find an array of bindings for extensions
+   */
+  getExtensionBindings(): Binding[] {
+    return this.context.find(`${this.name}.*`);
+  }
+
+  /**
+   * Get a map of extension bindings by the keys
+   */
+  getExtensionBindingMap(): {[name: string]: Binding} {
+    const extensions: {[name: string]: Binding} = {};
+    const bindings = this.context.find(`${this.name}.*`);
+    bindings.forEach(binding => {
+      extensions[binding.key] = binding;
+    });
+    return extensions;
+  }
+
+  /**
+   * Look up an extension binding by name
+   * @param name Name of the extension
+   */
+  getExtensionBinding(name: string): Binding {
+    const key = `${this.name}.${name}`;
+    const extensions = this.getExtensionBindingMap();
+    return extensions[key];
```

Comment:
I am proposing to simplify this method to the following:

```ts
return this.context.getBinding(key);
```

The current implementation builds a full map of all extension bindings, which is a preliminary pessimization IMO.

There is also a subtle difference handling of unknown extension names: the current proposal returns undefined, my proposal throws error `The key ${key} was not bound to any value.`.